apollo-router/src/services/supergraph_service.rs

```diff
@@ -229,6 +230,29 @@ async fn service_call(
             let is_deferred = plan.is_deferred(operation_name.as_deref(), &variables);
             let is_subscription = plan.is_subscription(operation_name.as_deref());
 
+            if let Some(batching) = context.private_entries.clone().lock().get::<Batching>() {
```

Comment:
Is clone mandatory here ?

Reply:
Yes, or the borrow checker gets angry that I'm moving out of `context` at line 249. It's a lightweight `Arc` clone.

Reply:
Oh yes indeed